lib/rubocop/cop/performance/reverse_each.rb

```diff
@@ -0,0 +1,41 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop is used to identify usages of `reverse.each` and
+      # change them to use `reverse_each` instead.
+      #
+      # @example
+      #   # bad
+      #   [].reverse.each
+      #
+      #   # good
+      #   [].reverse_each
+      class ReverseEach < Cop
+        MSG = 'Use `reverse_each` instead of `reverse.each`.'
+
+        def on_send(node)
+          return unless node.children.last == :each
```

Comment:
I'm not very fond of accessing the children array, as this hides the structure of the node. We usually prefer to destructure nodes using `*`.